pkg/controller/wildflyserver/wildflyserver_controller.go

```diff
@@ -302,12 +525,16 @@ func matches(container *v1.Container, envVar corev1.EnvVar) bool {
 }
 
 // statefulSetForWildFly returns a wildfly StatefulSet object
-func (r *ReconcileWildFlyServer) statefulSetForWildFly(w *wildflyv1alpha1.WildFlyServer) *appsv1.StatefulSet {
+func (r *ReconcileWildFlyServer) statefulSetForWildFly(w *wildflyv1alpha1.WildFlyServer, txnRecoverySecret *corev1.Secret) *appsv1.StatefulSet {
 	ls := labelsForWildFly(w)
+	delete(ls, labelWildflyOperatorInService)
 	replicas := w.Spec.Size
 	applicationImage := w.Spec.ApplicationImage
 	volumeName := w.Name + "-volume"
 
+	mgmtUser := string(txnRecoverySecret.Data["username"])
```

Comment:
maybe we should pass directly the user / password credentials in parameter instead of extracting them from the secret?